modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/api.ts

```diff
@@ -49,47 +52,82 @@ function getValidFields({
 
 	const properties: IProperties = schemas[schemaName]?.properties;
 
-	Object.keys(properties).map((propertyKey) => {
-		const propertyValue = properties[propertyKey];
+	properties &&
+		Object.keys(properties).map((propertyKey) => {
+			const propertyValue = properties[propertyKey];
 
-		if (INVALID_FIELDS.includes(propertyKey)) {
-			return;
-		}
+			if (INVALID_FIELDS.includes(propertyKey)) {
+				return;
+			}
 
-		if (propertyKey.includes(LOCALIZABLE_PROPERTY_SUFFIX)) {
-			return;
-		}
+			if (propertyKey.includes(LOCALIZABLE_PROPERTY_SUFFIX)) {
+				return;
+			}
 
-		const type = propertyValue.type;
+			const type = propertyValue.type;
+
+			if (type === EFieldType.ARRAY) {
+				if (propertyValue.items?.$ref) {
+					fields.push({
+						children: getValidFields({
+							contextPath: `${contextPath}${propertyKey}${FDS_ARRAY_FIELD_NAME_DELIMITER}`,
+							schemaName: propertyValue.items.$ref.replace(
+								/^.*\//,
+								''
+							),
+							schemas,
+						}),
+						label: propertyKey,
+						name: `${contextPath}${propertyKey}${FDS_ARRAY_FIELD_NAME_PARENT_SUFFIX}`,
+						type: type ? type : 'array',
+					});
+
+					return;
+				}
+			}
+
+			if (propertyValue.$ref) {
+				fields.push({
+					children: getValidFields({
+						contextPath: `${contextPath}${propertyKey}${FDS_NESTED_FIELD_NAME_DELIMITER}`,
+						schemaName: propertyValue.$ref.replace(/^.*\//, ''),
+						schemas,
+					}),
+					label: propertyKey,
+					name: `${contextPath}${propertyKey}${FDS_NESTED_FIELD_NAME_PARENT_SUFFIX}`,
+					type: type ? type : 'object',
+				});
+
+				return;
+			}
+
+			if (
```

Comment:
Enabling this type of extensions allow fine grain selection of common nested fields.
If it is a valid choice we could merge with the previous one (`if (propertyValue.$ref)`) as they handle items in the same way.